src/keybindings.c

```diff
@@ -782,6 +768,36 @@ static void load_kb(GeanyKeyGroup *group, GeanyKeyBinding *kb, gpointer user_dat
 }
 
 
+static void init_platform_kb(void)
+{
+#ifdef __APPLE__
+	gchar *configfile = g_build_filename(app->datadir, "keybindings_osx.conf", NULL);
```

Comment:
why loading a platform-named file instead of simply `keybindings.conf`, and install a different one (or none) depending on the platform? I mean, it doesn't make any sense to have an OSX keybinding file on Windows or GNU/Linux, nor it would to have a Windows one on OSX, etc.  Plus, it would make it easier to have specific keybindings on non-OSX in the future if wanted, without having to actually change the loading code.

The only "trick" would be to install an OSX-specific file on OSX, which should be reasonably easy to do.

Reply:
The only reason was my disgust towards automake/autoconf and not really knowing how to do it...

Reply:
I suggest something like 28ac94dac806b2fed92c3c7406557341c978fbcd (https://github.com/b4n/geany/tree/techee/osx_keybindings_followup).
To be tested if it properly detects OSX on an OSX system, but it should.

Reply:
Great! I knew you wouldn't resist the temptation to mess with m4 :-)

But the patch doesn't do exactly the right thing. The point of my patch was to keep Geany's keybindings.conf and only override the keybindings which have to change inside keybindings_osx.conf so it doesn't have to change every time some new keybinding is added and it's easier to see only the changes. See the calls of init_default_kb() followed by init_system_kb() which does the overrides.

So what I assumed you had in mind in your original comment was to keep `keybindings.conf` as it is now but then on install time copy only the keybindings file specific to the used platform and rename it to e.g. `keybindings_patform.conf` which would be referenced in the sources so there wouldn't have to be any ifdefs and only one version of platform-specific keybindings would be installed.

Also I think it would be best to keep the platform-specific keybinding files inside the data dir in sources and not to move it under osx as they will usually have to be updated together.

Now the question is how hard this would be to do this and whether it's worth the work...

Reply:
@techee hum, we don't have a system-wide keybindings file yet, so I don't really see the difference between what I did and what you suggest?  OK, if we wanted to have default keybindings in a system config file instead of hard-coded it would make sense (and it might be a good thing for the future), but unless I miss something we currently don't.

> Also I think it would be best to keep the platform-specific keybinding files inside the data dir in sources and not to move it under osx as they will usually have to be updated together.

Yeah… well the problem is that Autotools don't make it easy to rename a file when installing it, hence the reason why I put it in a sub-directory so it can have the same name for each platform.  But it's possible to do a rename, just not very nice.  (yes, it's a sometimes annoying limitation of Autotools -- although generally renaming files means you could not use a non-installed version of your software, but well)

Reply:
Meh, my bad, I haven't looked at the code for ages and completely forgot the defaults were in the code, not a config file.

The config file location probably isn't a big problem, especially when there's no Windows config file (which may never be needed) and no system-wide config file. So I guess your patch is good then.